docs/developers/build-cloud.md

```diff
@@ -0,0 +1,147 @@
+
+### Server
+
+- A server running Ubuntu 18.04 LTS
+  Can be hardware or virtual machine (tested on OpenStack, Amazon Web Services and Huawei Cloud).
+- At least 12GB disk size (including Ubuntu); better 15GB or more.
+  Note: After cleaning temporary files, QM occupies ~6GB of disk space.
+- Access to server via SSH key as user with `sudo` rights
+
+Security rules:
+
+- Port 22 open (for SSH access)
+- You may want to open further ports (optional):
+  - port 8888 to connect to Jupyter Notebook Servers (AiiDA lab)
+  - port 5000 to connect to the AiiDA REST API
+
+### Client (your computer)
+
+- [python](https://www.python.org/) >= 3.6
+- [git](https://git-scm.com)
+
+## Provisioning the server
+
+To get set up, run the following on your client (e.g. your laptop -- *not* on the server itself):
+
+```bash
+git clone https://github.com/marvel-nccr/quantum-mobile.git
+cd quantum-mobile
+```
+
+To allow ansible to connect to the server, you should adapt the `inventory.yml` file, to contain the correct connection details for the desired server.
+For example, to connect to an `aws` host, adapt:
+
+```yaml
+    aws:
+      cloud_platform: aws
+      ansible_host: 34.250.68.129 # change this
+      ansible_ssh_common_args: -i ./keys/quantum-mobile.pem -o StrictHostKeyChecking=no
+      ansible_user: ubuntu
+```
+
+with:
+
+- the IP of the server
+- the path to your private SSH key for connecting to the server
+- the user to connect as via SSH
+
+:::{seealso}
+The [ansible inventory](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html) and [playbook variables](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html) documentation.
+:::
+
+You can also "tune" the variables in `inventory.yml`, such as:
+
+- `vm_user`: the user for which to install the simulation environment (usually *not* the admin user you are connecting as)
+- `vm_memory`, `vm_cpus`
+- If you want to pre-load an SSH public key for the `vm_user` to connect with (as opposed to the admin key set above),
+  uncomment and set the `add_user_public_key` variable.
+
+Finally, to run the ansible playbook against the required host, use:
+
+```bash
+pip install tox
+tox -e py38-ansible -- --extra-vars "build_hosts=aws"
+```
+
+Your server should now be fully deployed and operational.
+
+:::{tip}
+The ansible automation steps are generally idempotent, meaning that if they have been previously run successfully, then they will be skipped in any subsequent runs.
+
+This means, if the build is interrupted for any reason, you can simply re-run the above command.
+:::
+
+You can log in to the server as the `vm_user` via the public SSH key you provided.
+
+## Saving the image
+
+Before creating an image from the disk volume of the server you provisioned:
+
+1. Remove unnecessary temporary build files:
+
+   ```bash
+   tox -e py38-ansible -- --tags quantum_espresso,qm_customizations,simulationbase,ubuntu_desktop --extra-vars "build_hosts=aws clean=true"
```

Comment:
Why should you need to understand what's going on under the hood?
I think you are just saying this because you are used to running the `ansible-playbook` command, which is understandable.
But IMO this is just an added improvement in ansible's goal for automation and repeatability, in that tox is also automating your python environment setup (plus installing the ansible roles).